src/DataUtils.cpp

```diff
@@ -0,0 +1,211 @@
+#include "NAS2D/DataUtils.h"
+
+#include "NAS2D/Xml/XmlAttribute.h"
+#include "NAS2D/Xml/XmlElement.h"
+
+#include <algorithm>
+#include <sstream>
+
+#if defined(_WIN32)
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#endif
+
+void NAS2D::Xml::validateXMLElement(
+	const NAS2D::Xml::XmlElement& element,
+	const std::string& name,
+	const std::string& requiredChildElements,
+	const std::string& requiredAttributes,
+	const std::string& optionalChildElements,
+	const std::string& optionalAttributes)
+{
+	if (name.empty())
+	{
+		// TODO: Make this a function with [[noreturn]] attribute
+#if defined(_WIN32)
+		std::ostringstream err_ss;
+		err_ss << "Xml Validation failed. Element name is required.";
+		::MessageBoxA(nullptr, err_ss.str().c_str(), "Fatal Error", MB_ICONERROR | MB_OK);
+#else
+		// TODO: Add non-windows version.
+#endif
+		exit(0);
+	}
+
+	const std::string elemName = element.value();
+	if (elemName != name)
+	{
+		// TODO: Make this a function with [[noreturn]] attribute
+#if defined(_WIN32)
+		std::ostringstream err_ss;
+		err_ss << "Xml Validation failed. Element name " << elemName << " does not match valid name " << name;
+		::MessageBoxA(nullptr, err_ss.str().c_str(), "Fatal Error", MB_ICONERROR | MB_OK);
```

Comment:
Simple string concatenation would probably be easier than using `ostringstream`. One caveat is there is no `operator+` that takes two `const char*`. At least one of the two operands must be `std::string` or similar. If that's not that case, you can always wrap a `const char *` with a `std::string` constructor call.

Reply:
It's also faster by a whole order of magnitude. Just tested `1'000'000` iterations of equivalent code using `std::ostringstream` vs `std::to_string`. `ostringstream` took around 3 seconds, `to_string` took around a third of a second.

Reply:
Oh hey, sweet. I didn't realize there would be that kind of performance gain.